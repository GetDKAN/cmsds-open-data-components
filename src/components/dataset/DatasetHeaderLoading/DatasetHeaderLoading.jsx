import React from 'react'

import './DatasetHeaderLoading.scss'

const DatasetHeaderLoading = () => (
  <div className='DatasetHeaderLoading' data-testid='DatasetHeaderLoading'>
    <div className='DatasetHeaderLoading__title skeleton-box' />
    <div className='DatasetHeaderLoading__desc-container'>
      <div className='DatasetHeaderLoading__desc_item skeleton-box' />
      <div className='DatasetHeaderLoading__desc_item skeleton-box' />
    </div>
    <div className='DatasetHeaderLoading__date skeleton-box' />
  </div>
)
DatasetHeaderLoading.displayName = 'DatasetHeaderLoading'
export default DatasetHeaderLoading
