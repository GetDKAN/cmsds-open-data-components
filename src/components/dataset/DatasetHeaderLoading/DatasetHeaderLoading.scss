.DatasetHeaderLoading {
  position: relative;
  margin-bottom: 24px;
  padding-bottom: 24px;

  &::after {
    content: "";
    width: 48px;
    height: 4px;
    background-color: #42e288;
    position: absolute;
    bottom: 0;
    left: 0;
  }

  > .DatasetHeaderLoading__title {
    margin: 1.61rem 0;
    height: 38px;
    width: 40%;
  }

  > .DatasetHeaderLoading__desc-container {
    > .DatasetHeaderLoading__desc_item {
      height: 18px;
      width: 100%;
      margin-bottom: 3px;
      display: block;

      &:last-child {
        width: 80%;
        margin-bottom: 0;
      }
    }
  }

  > .DatasetHeaderLoading__date {
    height: 16px;
    width: 30%;
    margin-top: 25px;
  }
}
