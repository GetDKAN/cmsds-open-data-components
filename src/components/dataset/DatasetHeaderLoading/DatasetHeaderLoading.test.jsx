import renderer from 'react-test-renderer'
import DatasetHeaderLoading from './DatasetHeaderLoading'

describe('DatasetHeaderLoading component.', () => {
  it('Matches snapshot.', async () => {
    const renderedDatasetHeaderLoading = renderer.create(
      <DatasetHeaderLoading />
    ).toJSON()

    expect(renderedDatasetHeaderLoading).toMatchSnapshot()
  })
})
