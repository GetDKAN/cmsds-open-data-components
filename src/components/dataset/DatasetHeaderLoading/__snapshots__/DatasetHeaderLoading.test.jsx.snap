// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DatasetHeaderLoading component. > Matches snapshot. 1`] = `
<div
  className="DatasetHeaderLoading"
  data-testid="DatasetHeaderLoading"
>
  <div
    className="DatasetHeaderLoading__title skeleton-box"
  />
  <div
    className="DatasetHeaderLoading__desc-container"
  >
    <div
      className="DatasetHeaderLoading__desc_item skeleton-box"
    />
    <div
      className="DatasetHeaderLoading__desc_item skeleton-box"
    />
  </div>
  <div
    className="DatasetHeaderLoading__date skeleton-box"
  />
</div>
`;
